internal/serve/httphandler/payments_handler.go

```diff
@@ -153,3 +155,58 @@ func (p PaymentsHandler) getPaymentsWithCount(ctx context.Context, queryParams *
 		return utils.NewResultWithTotal(totalPayments, payments), nil
 	})
 }
+
+type PatchPaymentStatusRequest struct {
+	Status string `json:"status"`
+}
+
+type UpdatePaymentStatusResponseBody struct {
+	Message string `json:"message"`
+}
+
+func (p PaymentsHandler) PatchPaymentStatus(w http.ResponseWriter, r *http.Request) {
+	var patchRequest PatchPaymentStatusRequest
+	err := json.NewDecoder(r.Body).Decode(&patchRequest)
+	if err != nil {
+		httperror.BadRequest("invalid request body", err, nil).Render(w)
+		return
+	}
+
+	// validate request
+	toStatus, err := data.ToPaymentStatus(patchRequest.Status)
+	if err != nil {
+		httperror.BadRequest("invalid status", err, nil).Render(w)
+		return
+	}
+
+	paymentManagementService := services.NewPaymentManagementService(p.Models, p.DBConnectionPool)
+	response := UpdatePaymentStatusResponseBody{}
+
+	ctx := r.Context()
+	paymentID := chi.URLParam(r, "id")
+
+	switch toStatus {
+	case data.CanceledPaymentStatus:
+		err = paymentManagementService.CancelPayment(ctx, paymentID)
+		response.Message = "Payment canceled"
+	default:
+		err = services.ErrPaymentStatusCantBeChanged
+	}
```

Comment:
Why not just use early return here?

```suggestion
	if toStatus != data.CanceledPaymentStatus {
		httperror.BadRequest(services.ErrPaymentStatusCantBeChanged.Error(), err, nil).Render(w)
		return
	}
	
	err = paymentManagementService.CancelPayment(ctx, paymentID)
	if err != nil {
		...
	}
```

Reply:
I used `switch` to patch other statuses if it's implemented in the future...